app/femr/business/services/system/SearchService.java

```diff
@@ -313,17 +317,22 @@ public SearchService(IRepository<IDiagnosis> diagnosisRepository,
                     .stream()
                     .filter(pp -> pp.getPatientPrescriptionReplacements() == null || pp.getPatientPrescriptionReplacements().size() == 0)
                     .map(pp -> itemModelMapper.createPrescriptionItem(
-                            pp.getId(),
-                            pp.getMedication().getName(),
-                            null,
-                            pp.getPhysician().getFirstName(),
-                            pp.getPhysician().getLastName(),
-                            pp.getMedicationAdministration(),
-                            pp.getAmount(),
-                            pp.getMedication(),
-                            null,
-                            pp.isCounseled()
 
+                        pp.getId(),
+                        pp.getMedication().getName(),
+                        null,
+                        pp.getPhysician().getFirstName(),
+                        pp.getPhysician().getLastName(),
+                        pp.getMedicationAdministration(),
+                        pp.getAmount(),
+                        pp.getMedication(),
+                        pp.getMedication()
+                            .getMedicationInventory()
+                            .stream()
+                            .filter( i -> i.getMissionTrip().getId() == pp.getPatientEncounter().getMissionTrip().getId() )
+                            .findFirst()
+                            .get(),
+                        pp.isCounseled()
```

Comment:
@kevinzurek: So when fetching the inventory for a certain medication, we get back the inventories from all trips in the system - Obviously we only want the inventory for the current user's trip. What I did above works, but I feel like there is a better way. Am I missing something?